### PR TITLE
Remove unused variables and hide normal participants

### LINKED ISSUE #45: Custom filter for participants with 1+ non-normal value
# Test notes
* A checkbox appears beneath the x-axis dropdown labeled _Inliers_ that toggles the display of participants in each chart without values outside the normal range.

### LINKED ISSUE #79: Limit input data to only those variables required
Limit input data to only those variables required: measure, ID, visit, visit order, visit day, and results, and optionally measure units and limits of normal.

Drop extraneous variables before drawing charts.

Warn user of rendering limitations due to large data files.

The goal of this issue is to improve performance.

# Test notes
* In the console tab of the developer console (press F12 to reveal) enter `d3.select('.line').datum().values[0].values.raw[0]`.  The variables returned should be limited to these:
  * STNRHI
  * STNRLO
  * STRESN
  * STRESU
  * TEST
  * USUBJID
  * VISIT
  * VISITNUM
  * abnormal
  * abnormalID
  * brushed
  * measure_unit
  * unscheduled
  * and any additional time variables specified in the settings, e.g. DY

### PR DESCRIPTION
**Pull request by samussiah**
# Issues
Closes #45 
Closes #79

### CODE REVIEW

File: src/defaultSettings.js [hunk 1]
```diff
@@ -19,6 +11,14 @@ export default {
...
         },
+        {
+            value_col: 'DY',
+            type: 'linear',
+            order: null,
+            label: 'Study Day',
+            rotate_tick_labels: false,
+            vertical_space: 0
+        },
         {
```
Comment (jwildfire) [inline src/defaultSettings.js:22]: Guessing we're going to remove numeric visit axis in a later PR here ...

File: src/index.js [hunk 2]
```diff
@@ -1,3 +1,7 @@
+//Utility polyfills
+import './util/object-assign';
```
Comment (jwildfire) [inline src/index.js:2]: Are the new imports being used?

File: src/init/defineData/deriveVariables.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    this.data.raw.forEach(d => {
+        d.brushed = false;
+        if (d[this.config.unit_col])
+            d.measure_unit = `${d[this.config.measure_col]} (${d[this.config.unit_col]})`;
+        else d.measure_unit = d[this.config.measure_col];
+
+        const lo =
+                d[this.config.lln_col] !== undefined
+                    ? +d[this.config.value_col] < +d[this.config.lln_col]
+                    : false,
```
Comment (jwildfire) [inline src/init/defineData/deriveVariables.js:11]: We should consider imputing  upper and lower limits when none are given in the data. Would be nice to provide a control that lets users define outliers using either Standard deviations or Quantiles. We can handle this in a separate issue.

File: src/init/defineData/removeVariables.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { set, merge } from 'd3';
```
Comment (jwildfire) [inline src/init/defineData/removeVariables.js:1]: Looks good. This addresses #79, so I'm going to close #86 without merging and add #79 to this PR.

File: src/util/array-find.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+if (!Array.prototype.find) {
```
Comment (jwildfire) [inline src/util/array-find.js:1]: Find not supported in old browsers?

**Review by jwildfire — COMMENTED** [review]
Looks great. Approving.

**Review by jwildfire — APPROVED** [review]

**Comment by jwildfire** [comment]
@samussiah both issues need some feature tests.

File: src/defaultSettings.js (hunk 1 above)
Comment (samussiah) [reply src/defaultSettings.js:22]: Yup.

File: src/index.js (hunk 2 above)
Comment (samussiah) [reply src/index.js:2]: `Object.assign()` just once, `Array.find()` everywhere.

File: src/util/array-find.js (hunk 5 above)
Comment (samussiah) [reply src/util/array-find.js:1]: Not in IE :crying_cat_face:

**Comment by brittsikora** [comment]
This branch isn't rendering in Firefox or Chrome.  

hide-normal-participants

I removed the settings and it renders, but it looks jacked.

hide-normal-participants2

**Comment by brittsikora** [comment]
Pass

Inliers option works as expected and only required variables show up in the console

**Review by brittsikora — APPROVED** [review]

**Pull request merged**